lib/plugins/aws/provider/awsProvider.js

```diff
@@ -187,6 +187,13 @@ class AwsProvider {
     impl.addEnvironmentProfile(result, `AWS_${stageUpper}`);
 
     result.region = this.getRegion();
+
+    if (this.serverless.service.provider.deploymentBucketObject &&
+        this.serverless.service.provider.deploymentBucketObject.serverSideEncryption &&
+        this.serverless.service.provider.deploymentBucketObject.serverSideEncryption ===
+         'aws:kms') {
+      result.signatureVersion = 'v4';
+    }
```

Comment:
> regarding the sdk version, it is set up as "^2.7.13" which pulls in the latest version 2.61.0, right?

@Athulus we're using a shrinkwrap file where we lock down the versions. I'd recommend to update the version in the `package.json` file and run `npm shrinkwrap`. This should do the trick.

> is there a cleaner way to check for existence? this seems a little verbose.

Looks good. You could extract this into an own variable so that you don't have to rewrite `this.serverless.service...` over and over again.

> sorry if that is a stupid question or obviously wrong.

No worries! Glad that you're brave enough to jump right into this!